compiler+runtime/src/jank/clojure/string.jank

```diff
@@ -4,320 +4,318 @@
+;  ;        (recur (unchecked-dec index))
+;  ;        (.. s (subSequence 0 index) toString)))))
+;  ;
+;  ;(defn trim-newline
+;  ;  "Removes all trailing newline \\n or return \\r characters from
+;  ;  string.  Similar to Perl's chomp."
+;  ;  {:added "1.2"}
+;  ;  [s]
+;  ;  (loop [index (.length s)]
+;  ;    (if (zero? index)
+;  ;      ""
+;  ;      (let [ch (.charAt s (dec index))]
+;  ;        (if (or (= ch \newline) (= ch \return))
+;  ;          (recur (dec index))
+;  ;          (.. s (subSequence 0 index) toString))))))
+;  
+;  (defn blank?
+;    "True if s is nil, empty, or contains only whitespace."
+;    [s]
+;    (native/raw "auto const s_str(runtime::to_string(~{ s }));
+;                __value = make_box(s_str.is_blank());"))
+;  
+;  ;(defn escape
+;  ;  "Return a new string, using cmap to escape each character ch
+;  ;  from s as follows:
+;  ;
+;  ;  If (cmap ch) is nil, append ch to the new string.
+;  ;  If (cmap ch) is non-nil, append (str (cmap ch)) instead."
+;  ;  {:added "1.2"}
+;  ;  [s cmap]
+;  ;  (loop [index (int 0)
+;  ;         buffer (StringBuilder. (.length s))]
+;  ;    (if (= (.length s) index)
+;  ;      (.toString buffer)
+;  ;      (let [ch (.charAt s index)]
+;  ;        (if-let [replacement (cmap ch)]
+;  ;          (.append buffer replacement)
+;  ;          (.append buffer ch))
+;  ;        (recur (inc index) buffer)))))
+;  ;
+;  ;(defn index-of
+;  ;  "Return index of value (string or char) in s, optionally searching
+;  ;  forward from from-index. Return nil if value not found."
+;  ;  {:added "1.8"}
+;  ;  ([s value]
+;  ;   (let [result (if (instance? Character value)
+;  ;                  (.indexOf (.toString s) (.charValue value))
+;  ;                  (.indexOf (.toString s) value))]
+;  ;     (if (= result -1)
+;  ;       nil
+;  ;       result)))
+;  ;  ([s value from-index]
+;  ;   (let [result (if (instance? Character value)
+;  ;                  (.indexOf (.toString s) (.charValue value) (unchecked-int from-index))
+;  ;                  (.indexOf (.toString s) value (unchecked-int from-index)))]
+;  ;     (if (= result -1)
+;  ;       nil
+;  ;       result))))
+;  ;
+;  ;(defn last-index-of
+;  ;  "Return last index of value (string or char) in s, optionally
+;  ;  searching backward from from-index. Return nil if value not found."
+;  ;  {:added "1.8"}
+;  ;  ([s value]
+;  ;   (let [result (if (instance? Character value)
+;  ;                  (.lastIndexOf (.toString s) (.charValue value))
+;  ;                  (.lastIndexOf (.toString s) value))]
+;  ;     (if (= result -1)
+;  ;       nil
+;  ;       result)))
+;  ;  ([s value from-index]
+;  ;   (let [result (if (instance? Character value)
+;  ;                  (.lastIndexOf (.toString s) (.charValue value) (unchecked-int from-index))
+;  ;                  (.lastIndexOf (.toString s) value (unchecked-int from-index)))]
+;  ;     (if (= result -1)
+;  ;       nil
+;  ;       result))))
+;  
 (defn starts-with?
```

Comment:
In `core.jank`, the pattern I've noticed is the use a def here instead of a defn. I feel like it was intentional to avoid the extra fn, but not sure.

Reply:
You're correct. We already have a lot of indirection. We don't want even more.